File: CSharp/DP.Domain/Samples/Interpreter/VipExpression.cs
namespace DP.Domain.Samples.Interpreter
{
    /// <summary>
    /// TerminalExpression
    /// </summary>
    public class VipExpression : IExpression
    {
        public void Interpret(Context context)
        {
            context.Output.Vip.CardNo = context.Input.Substring(24, 6).Trim();
            context.Output.Vip.BonusPoints = int.Parse(context.Input.Substring(30, 8).Trim());
        }
    }
}

File: Python/Samples/Interpreter/VipExpression.py
from decimal import Decimal 
from Interpreter_models import PayData, Vip, Store, Context
from Expression import Expression

class VipExpression(Expression):
     def interpret(self,context=Context):
         context.output.vip.cardNo = context.input[24:30].strip()
         context.output.vip.bonusPoints = context.input[30:38].strip()